### PR TITLE
Refactored User Controller

### LINKED ISSUE #205: Refactor User Controller
## Refactor User Controller

1. Remove Authentication
2. Remove Tests
3. Rewrite Controller into Repository Pattern
4. Write new tests 

**Things To Consider with Testing**

The tests should be written to the Controller and Repository specifically. 
Which should ease the testing process significantly. When testing the Controller it is possible to mock the Repository to fit the test.

Here's some information regarding the Repository Pattern, which should be implemented in the Controllers. It is crucial that you read it all and understand. Just click the Link: [Repository Pattern Docs](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/infrastructure-persistence-layer-design)

Also, currently the testing is being done with a mocked database context, which is both tedious and a mess, the solution to this is using EF's in memory database to do local testing. 
There will of course come testing later on release when the CICD Pipeline runs (hopefully soon), but it still necessary to have quality unit tests.
This can be studied in further detail using the link above and google, which as always is your friend.

**Regarding Removing Authentication**

This is of course temporary, and there will come a solution to replace it in due time.

**Regarding Transparency**

The current controller system is outdated and poorly executed, therefore we the process group have agreed to change it, we hope you agree with the choice of pattern. If you have any concerns, tell your process representative right away, since this task is done for all controllers and consistency is key.

### PR DESCRIPTION
**Pull request by fatimaosman99**
# Description

- Removed Authentication from the user controller
- Implemented repository pattern and made repositories that encapsulate query related methods
- Unit tested the repositories
- Unit tested the user controller

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes #205 

## Type of change

*Please delete options that are not relevant.*

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Refactoring

# How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration*

- [x] Unit tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if necessary
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by pannaAwesome — APPROVED** [review]

File: GirafRest/Controllers/UserController.cs [hunk 1]
```diff
@@ -338,41 +333,22 @@ public async Task<ActionResult> AddUserResource(string id, [FromBody] ResourceId
...
-
-            //Check that the currently authenticated user owns the resource
-            var curUsr = await _giraf.LoadBasicUserDataAsync(HttpContext.User);
-            var resourceOwnedByCaller = await _giraf.CheckPrivateOwnership(resource, curUsr);
-            if (!resourceOwnedByCaller)
-                return StatusCode(StatusCodes.Status403Forbidden,
-                    new ErrorResponse(ErrorCode.NotAuthorized, "User does not own resource"));
-
-            //Check if the target user already owns the resource
-            if (user.Resources.Any(ur => ur.PictogramKey == resourceIdDTO.Id))
```
Comment (Drejer1) [inline GirafRest/Controllers/UserController.cs:369]: Hvorfor er denne if slettet? Kan ikke se du har erstattet den noget andet sted?

Comment (fatimaosman99) [reply GirafRest/Controllers/UserController.cs:369]: Godt fanget. Jeg tror det er fordi jeg troede den var relateret til authentication if-statement. Jeg sætter den ind igen.

File: GirafRest/Interfaces/IGirafService.cs [hunk 2]
```diff
@@ -28,14 +28,24 @@ GirafDbContext _context
...
+            set;
         }
 
+
+        /// <summary>
+        /// Reads an image from the current request's body and return it as a byte array.
+        /// </summary>
+        /// <param name="bodyStream">A byte-stream from the body of the request.</param>
+        /// <returns>The image found in the request represented as a byte array.</returns>
+        Task<byte[]> ReadRequestImage(Stream bodyStream);
```
Comment (Drejer1) [inline GirafRest/Interfaces/IGirafService.cs:47]: Hvorfor er der tilføjet ting til IGirafService? bruger du den til noget?

File: GirafRest/Repositories/WeekRepository.cs [hunk 3]
```diff
@@ -16,17 +16,10 @@ public WeekRepository(GirafDbContext context) : base(context)
         {
         }
```
Comment (Drejer1) [inline GirafRest/Repositories/WeekRepository.cs:17]: Der er ændret på tin i WeekRepository? troede ikke du havde haft noget med den at gøre?

**Review by Drejer1 — CHANGES_REQUESTED** [review]
Se kommentare

File: GirafRest/Interfaces/IGirafService.cs (hunk 2 above)
Comment (fatimaosman99) [reply GirafRest/Interfaces/IGirafService.cs:47]: Jeg ved ikke hvorfor den er blevet tilføjet men jeg kan se det bliver brugt i pictogram controller.

File: GirafRest/Repositories/WeekRepository.cs (hunk 3 above)
Comment (fatimaosman99) [reply GirafRest/Repositories/WeekRepository.cs:17]: Det er fordi der var nogle merge conflicts mellem min branch og develop så der er kommet ny ændringer ind fra develop. Men jeg tilføjer de slettede kommentar & xml documentation.

**Review by Drejer1 — APPROVED** [review]

**Pull request merged**